Mathlib/Tactic/StacksAttribute.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2024 Dmaiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dmaiano Testa
+-/
+import Lean.Parser.Syntax
+
+/-!
+The `stacks` attribute.
+-/
+
+open Lean
+
+namespace Mathlib.Stacks
+
+/--
+The syntax for a Stacks tag: it is an optional number followed by an optional identifier.
+This allows `044Q3` and `GH3F6` as possibilities.
+-/
+declare_syntax_cat stackTags
+
+@[inherit_doc Parser.Category.stackTags]
+syntax (num)? (ident)? : stackTags
+
+/-- The `stacks` attribute.
+Use it as `@[stacks TAG "Optional comment"]`.
+The `TAG` is mandatory.
+-/
+syntax (name := stacks) "stacks " (stackTags)? (str)? : attr
+
+initialize Lean.registerBuiltinAttribute {
+  name := `stacks
+  descr := "Apply a Stacks project tag to a theorem."
+  add := fun _decl stx _attrKind => Lean.withRef stx do
+    match stx with
+      | `(attr| stacks $_:stackTags $_:str) => return
```

Comment:
Can we validate the `stackTag` here?

Reply:
I just added a check that there are no spaces in the tag itself.  By "validate" you mean "make sure that such a tag exists"?  That should probably be done by a linter, right?

Reply:
Sorry. What something like
```lean
syntax num | lit : stacksTagAtom
syntax stacksAtom stacksAtom stacksAtom stacksAtom stacksAtom : stacksTag
```
Is there some uppercase Latin character syntax category? Do we really have to write out `stacksAtom` 5 times?

Reply:
Oh, the length 5 check can be done with the substring and also the uppercase check.

Reply:
I have a slight preference for not using the double quotes, which a `str` would want.  If there is a way of parsing a string without quotes, then I would probably prefer that!